rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/data_definitions.kt

```diff
@@ -756,7 +756,14 @@ fun decodeFromDS(value: String, digits: Int, scale: Int): BigDecimal {
     }
     number = sign + number
     return try {
-        value.toBigDecimal()
+        val builder = StringBuilder()
+        val valueWithoutSeparator = value.replace(".", "").replace(",", "")
+        val decimalSlice = valueWithoutSeparator.substring(valueWithoutSeparator.length - scale)
+        builder
+            .append(valueWithoutSeparator.substring(0, valueWithoutSeparator.length - decimalSlice.length))
+            .append(".")
+            .append(decimalSlice)
+        builder.toString().toBigDecimal()
     } catch (e: Exception) {
         number.toBigDecimal()
```

Comment:
I don't like this thing, have you investigated in order to understand why?

Reply:
I'm agree but I need to discuss to it. I thinks that String to Number and DS to Number are entire managed in wrong way. I might be wrong.

In the function there are a number of operations on bytes. They are really necessary?

Reply:
But if you remove the catch block what happens?